File: engine/src/main/java/client/SkillEntry.java
package client;

public class SkillEntry {

   private int masterLevel;
   private byte skillLevel;
   private long expiration;

   public SkillEntry(byte skillLevel, int masterLevel, long expiration) {
      this.skillLevel = skillLevel;
      this.masterLevel = masterLevel;
      this.expiration = expiration;
   }

   public int getMasterLevel() {
      return masterLevel;
   }

   public byte getSkillLevel() {
      return skillLevel;
   }

   public long getExpiration() {
      return expiration;
   }

   @Override
   public String toString() {
      return skillLevel + ":" + masterLevel;
   }
}

File: engine/model/src/main/scala/client/SkillEntry.scala
package client

class SkillEntry(private var _skillLevel: Byte, private var _masterLevel: Int, private var _expiration: Long) {
  def skillLevel: Byte = _skillLevel

  def masterLevel: Int = _masterLevel

  def expiration: Long = _expiration

  override def toString: String = skillLevel + ":" + masterLevel
}
